### PR TITLE
create-react-app is dead, long live Vite.

### PR DESCRIPTION
**Pull request by torhovland**

### CODE REVIEW

**Comment by aspiwack** [comment]
Can I ask the motivation for such a change?

**Comment by torhovland** [comment]
Well, it's in the title, but sure 🙂

You see, despite its name `create-react-app` is not just used when creating an app. It's really a framework for running a React app. And it introduces a number of dependencies, which we won't be able to upgrade as `create-react-app` stops being maintained.

Also, this switch of tools gives us a more performant development experience.

[More background here.](https://github.com/reactjs/react.dev/pull/5487#issuecomment-1409720741)

File: web-playground/src/vite-env.d.ts
Comment (ErinvanderVeen) [inline web-playground/src/vite-env.d.ts:1]: What does this do?

File: web-playground/jest.config.js
Comment (ErinvanderVeen) [inline web-playground/jest.config.js:6]: Needs a newline

File: web-playground/.gitignore
Comment (ErinvanderVeen) [inline web-playground/.gitignore:24]: I don't like to have editor-specific configuration in the repository if I can help it.

**Review by ErinvanderVeen — CHANGES_REQUESTED** [review]

File: web-playground/.gitignore
Comment (torhovland) [reply web-playground/.gitignore:24]: This is just the default `.gitignore` that Vite created. I can of course change it, but I actually like letting team members add whatever they like to `.gitignore`, so they can use the tools they want with no risk of cluttering up the repo with files that don't belong there.

I even go one step further, and like to have shared config for various editors too, when a project demands certain config settings (which means I wouldn't actually ignore `.vscode/*`).

File: web-playground/src/vite-env.d.ts
Comment (torhovland) [reply web-playground/src/vite-env.d.ts]: Hmm, looks like we don't need this file. Removed.

File: web-playground/jest.config.js
Comment (torhovland) [reply web-playground/jest.config.js]: Fixed.

Comment (ErinvanderVeen) [reply web-playground/jest.config.js]: It wasn't quite, so I force pushed a fix.

**Review by ErinvanderVeen — APPROVED** [review]

**Comment by torhovland** [comment]
@ErinvanderVeen I was going to wait for CI before merging, but oh well 😄

File: web-playground/jest.config.js
Comment (torhovland) [reply web-playground/jest.config.js]: Oh, I checked in indenting on the last line 🤦‍♂️

Maybe I should start using pre-commit hooks.

**Pull request merged**
